src/serialization.c

```diff
@@ -279,4 +236,161 @@ RRStatus RaftRedisCommandArrayDeserialize(RaftRedisCommandArray *target, const v
     return RR_OK;
 }
 
+RRStatus RaftRedisDeserializeImport(ImportKeys * target, const void *buf, size_t buf_size)
+{
+    const char *p = buf;
+    int n;
+
+    FreeImportKeys(target);
+
+    /* Read term */
+    size_t term;
+    if ((n = decodeInteger(p, buf_size, '*', &term)) < 0 || !term) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
+    target->term = term;
+
+    /* Read migration_session_key */
+    size_t migration_session_key;
+    if ((n = decodeInteger(p, buf_size, '*', &migration_session_key)) < 0) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
+    target->migration_session_key = (unsigned long long) migration_session_key;
+
+    /* Read number of keys serialized in import entry */
+    size_t num_keys;
+    if ((n = decodeInteger(p, buf_size, '*', &num_keys)) < 0 || !num_keys) {
+        return RR_ERROR;
+    }
+    p += n; buf_size -= n;
+    target->num_keys = num_keys;
+
+    target->key_names = RedisModule_Calloc(num_keys, sizeof(RedisModuleString*));
+    target->key_serialized = RedisModule_Calloc(num_keys, sizeof(RedisModuleString*));
 
+    for (size_t i = 0; i < num_keys; i++) {
+        if ((n = decodeString(p, buf_size, &target->key_names[i])) < 0 || !target->key_names[i]) {
+            return RR_ERROR;
+        }
+        p += n; buf_size -= n;
+        if ((n = decodeString(p, buf_size, &target->key_serialized[i])) < 0 || !target->key_serialized[i]) {
+            return RR_ERROR;
+        }
+        p += n; buf_size -= n;
+    }
```

Comment:
```suggestion
        n = decodeString(p, buf_size, &target->key_names[i]);
        if (n  < 0 || !target->key_names[i]) {
            return RR_ERROR;
        }
        p += n; 
        buf_size -= n;

        n = decodeString(p, buf_size, &target->key_serialized[i]);
        if (n < 0 || !target->key_serialized[i]) {
            return RR_ERROR;
        }
        p += n; 
        buf_size -= n;
```

Please consider taking function call out of if check. The longer it gets, it becomes harder to read. Also, consider this for other usages in serialization code if it applies.